rust/crd/src/lib.rs

```diff
@@ -109,32 +112,45 @@ pub enum Error {
     )
 )]
 pub struct HiveClusterSpec {
-    /// General Hive metastore cluster settings
+    // no doc - docs in HiveClusterConfig struct.
     pub cluster_config: HiveClusterConfig,
-    /// Cluster operations like pause reconciliation or cluster stop.
+
+    // no doc - docs in ClusterOperation struct.
     #[serde(default)]
     pub cluster_operation: ClusterOperation,
-    /// The Hive metastore image to use
+
+    // no doc - docs in ProductImage struct.
     pub image: ProductImage,
+
+    // no doc - docs in Role struct.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub metastore: Option<Role<MetaStoreConfigFragment>>,
 }
 
+/// Hive metastore settings that affect all roles and role groups.
+/// The settings in the cluster config settings are settings that do not need to be configurable at role level.
```

Comment:
Not sure if going for `cluster config` or `clusterConfig`?
```suggestion
/// The settings in the cluster config are cluster wide settings that do not need to be configurable at role or role group level.
```